test/extensions/filters/http/oauth2/oauth_test.cc

```diff
@@ -79,12 +79,16 @@ TEST_F(OAuth2ClientTest, RequestAccessTokenSuccess) {
   mock_response->body().add(json);
 
   EXPECT_CALL(cm_.thread_local_cluster_.async_client_, send_(_, _, _))
-      .WillRepeatedly(
-          Invoke([&](Http::RequestMessagePtr&, Http::AsyncClient::Callbacks& cb,
-                     const Http::AsyncClient::RequestOptions&) -> Http::AsyncClient::Request* {
-            callbacks_.push_back(&cb);
-            return &request_;
-          }));
+      .WillRepeatedly(Invoke([&](Http::RequestMessagePtr& message, Http::AsyncClient::Callbacks& cb,
+                                 const Http::AsyncClient::RequestOptions&)
+                                 -> Http::AsyncClient::Request* {
+        EXPECT_FALSE(message->headers().get(Http::CustomHeaders::get().Accept).empty());
```

Comment:
I think this is fine, yeah. To make it complete, should we add checks for all the expected headers too here (including content-type, etc)?

Reply:
+1 for checking `FormUrlEncoded` ContentType.

Reply:
Added a check for content-type too here. I don't think this sets any other headers